integrations/dalle/src/index.ts

```diff
@@ -0,0 +1,78 @@
+import * as botpress from '.botpress'
+import axios from 'axios'
+import { buildApiData, validateResponse, getApiConfig } from './client'
+import { TGenerateImageOutput } from './types'
+
+class NotImplementedError extends Error {
+  constructor() {
+    super('Not implemented')
+  }
+}
+
+export default new botpress.Integration({
+  register: async () => { },
+  unregister: async () => { },
+  actions: {
+    generateImage: async ({ ctx, input, logger }): Promise<TGenerateImageOutput> => {
+
+      logger.forBot().info('Generating Image')
+
+      const { apiUrl, headers } = getApiConfig(ctx)
+      const data = buildApiData(input)
+
+      try {
+
+        const response = await axios.post(apiUrl, data, { headers })
+        validateResponse(response)
+
+        const image = response.data.data[0].url
+        const createdDate = response.data.created.toString()
+
+        return { url: image, createdDate }
+      } catch (error) {
+        logger.forBot().error('Error creating image:', error)
+        return { url: '', createdDate: Date.now().toString()}
+      }
+    }
+  },
+  channels: {
+    channel: {
```

Comment:
**note:** you'll be able to remove this once you remove the messaging channel form the definition